src/flash-source-buffer.js

```diff
@@ -382,17 +392,26 @@ export default class FlashSourceBuffer extends videojs.EventTarget {
 
       if (startTag.pts >= targetPts) {
         filteredVideoTags.push(startTag);
-      } else if (startTag.keyFrame) {
-        let nextIndex = startIndex + 1;
+      } else if (startTag.metaDataTag || startTag.keyFrame) {
+        let keyFrameIndex = getKeyFrameIndex(startTag, startIndex);
```

Comment:
verifying the index is mostly a sanity check as I think its possible (but I have not confirmed) that the two metadata tags could be followed by a non key frame. (I think if the segment does not start with a key frame. Here is the related mux.js code snippet

```
// Check if keyframe and the length of tags.
// This makes sure we write metadata on the first frame of a segment.
if (config && track && track.newMetadata &&
   (frame.keyFrame || tags.length === 0)) {
  // Push extra data on every IDR frame in case we did a stream change + seek
  tags.push(metaDataTag(config, frame.dts).finalize());
  tags.push(extraDataTag(track, frame.dts).finalize());
  track.newMetadata = false;
}
```

if `track.newMetadata === true && tags.length === 0 && frame.keyFrame === false`, the first tag being added would have the two metadata tags before it, but not be a key frame. By having that check, we wont treat a portion of tags as a GOP when they shouldn't be.